pkg/microservice/aslan/core/environment/service/environment.go

```diff
@@ -1689,6 +1705,41 @@ func GetHelmChartVersions(productName, envName string, log *zap.SugaredLogger) (
 	return helmVersions, nil
 }
 
+func DeleteProductServices(envName, productName string, serviceNames []string, log *zap.SugaredLogger) (err error) {
+	productInfo, err := commonrepo.NewProductColl().Find(&commonrepo.ProductFindOptions{Name: productName, EnvName: envName})
+	if err != nil {
+		log.Errorf("find product error: %v", err)
+		return err
+	}
+
+	for serviceGroupIndex, serviceGroup := range productInfo.Services {
+		var group []*commonmodels.ProductService
+		for _, service := range serviceGroup {
+			if !util.InArray(service.ServiceName, serviceNames) {
+				group = append(group, service)
+			}
+		}
+		err := commonrepo.NewProductColl().UpdateGroup(envName, productName, serviceGroupIndex, group)
+		if err != nil {
+			log.Errorf("update product error: %v", err)
+			return err
+		}
+	}
+	kubeClient, err := kubeclient.GetKubeClient(config.HubServerAddress(), productInfo.ClusterID)
+	if err != nil {
+		return e.ErrUpdateEnv.AddErr(err)
+	}
+	for _, name := range serviceNames {
+		selector := labels.Set{setting.ProductLabel: productName, setting.ServiceLabel: name}.AsSelector()
+		err = commonservice.DeleteResourcesAsync(productInfo.Namespace, selector, kubeClient, log)
+		if err != nil {
+			//删除失败仅记录失败日志
+			log.Errorf("delete resource of service %s error:%v", name, err)
+		}
```

Comment:
when deleting services in environment, should we set the status of prodcut to 'updating' just like an updating operation, this can also avoid concurrent product operation(mutexAutoUpdate). And the error can be store in product.error.